otter/test/convergence/test_transforming.py

```diff
@@ -87,9 +96,129 @@ def test_get_limits_conf(self):
         self.assertEqual(limits, {CreateServer: 100, CreateStack: 10})
 
 
+class FilterMutatingCLBTests(SynchronousTestCase):
+    """
+    Tests for :func:`filter_clb_mutating_types`
+    """
+
+    sample_steps = [
+        CreateServer(server_config=pmap({"name": "server"})),
+        CreateServer(server_config=pmap({"name": "server2"})),
+        DeleteServer(server_id="abc"), DeleteServer(server_id="def"),
+        AddNodesToCLB(
+            lb_id='5',
+            address_configs=s(('1.1.1.1',
+                               CLBDescription(lb_id='5', port=80)))),
+        AddNodesToCLB(
+            lb_id='5',
+            address_configs=s(('1.2.3.4',
+                               CLBDescription(lb_id='5', port=80)))),
+        AddNodesToCLB(
+            lb_id='6',
+            address_configs=s(('2.1.1.1',
+                               CLBDescription(lb_id='6', port=80)))),
+        AddNodesToCLB(
+            lb_id='6',
+            address_configs=s(('2.2.3.4',
+                               CLBDescription(lb_id='6', port=80)))),
+        RemoveNodesFromCLB(lb_id='5', node_ids=s('1')),
+        RemoveNodesFromCLB(lb_id='5', node_ids=s('2')),
+        RemoveNodesFromCLB(lb_id='6', node_ids=s('3')),
+        RemoveNodesFromCLB(lb_id='6', node_ids=s('4')),
+        ChangeCLBNode(
+            lb_id='5',
+            node_id='1',
+            condition=CLBNodeCondition.DRAINING,
+            weight=50,
+            type=CLBNodeType.PRIMARY),
+        ChangeCLBNode(
+            lb_id='5',
+            node_id='2',
+            condition=CLBNodeCondition.ENABLED,
+            weight=10,
+            type=CLBNodeType.PRIMARY),
+        ChangeCLBNode(
+            lb_id='6',
+            node_id='3',
+            condition=CLBNodeCondition.DRAINING,
+            weight=10,
+            type=CLBNodeType.PRIMARY),
+        ChangeCLBNode(
+            lb_id='6',
+            node_id='4',
+            condition=CLBNodeCondition.ENABLED,
+            weight=15,
+            type=CLBNodeType.SECONDARY)
+    ]
+
+    @given(st.lists(st.sampled_from(sample_steps)))
+    def test_only_one_mutating_type(self, steps):
+        """
+        Test that after filtering only one CLB mutating type steps exist per
+        CLB. Also ensure that non-CLB steps remain intact.
+        """
+        steps = pbag(steps)
+
+        clb_types = (AddNodesToCLB, RemoveNodesFromCLB, ChangeCLBNode)
+        lb_steps, no_lb_steps = partition_bool(lambda s: type(s) in clb_types,
+                                               steps)
+        steps_by_id = groupby(lambda s: s.lb_id, lb_steps)
+
+        filtered = filter_clb_mutating_types(steps)
+
+        filtered_steps_by_id = groupby(lambda s: getattr(s, "lb_id", "no"),
+                                       filtered)
+        for lb_id, lb_steps in steps_by_id.items():
```

Comment:
I agree. When I started writing regular unit tests there seemed to be too many cases which tested the same property: "Are the steps returned of one type per CLB?". So I thought of using hypothesis instead. However, testing that property for any kind of input turned out to be more work than expected. In any case, now that the function will be very simple it can have regular unit tests. 

(I was hoping to sneak in hypothesis with this PR but maybe some other time)